core/tee/tee_svc_storage.c

```diff
@@ -455,22 +434,35 @@ TEE_Result tee_svc_storage_obj_open(uint32_t storage_id, void *object_id,
 
 	res = tee_svc_storage_read_head(sess, o);
 	if (res != TEE_SUCCESS) {
+#ifdef CFG_ENC_FS
+		if (res != TEE_ERROR_ITEM_NOT_FOUND) {
+			tee_obj_add(sess->ctx, o);
+			res = tee_obj_verify(sess, o);
+			goto exit;
+		}
+#endif
 		free(o);
-		goto exit;
+		goto err;
 	}
 
 	file = tee_svc_storage_create_filename(sess, object_id,
 					       object_id_len, false);
 	if (file == NULL) {
 		res = TEE_ERROR_OUT_OF_MEMORY;
-		goto exit;
+		goto err;
+	}
+
+	err = tee_file_ops.access(file, TEE_FS_F_OK);
+	if (err) {
+		/* file not found */
+		res = TEE_ERROR_STORAGE_NOT_AVAILABLE;
+		goto err;
 	}
 
 	fd = tee_file_ops.open(file, fs_flags);
-	free(file);
 	if (fd < 0) {
-		res = TEE_ERROR_ITEM_NOT_FOUND;
-		goto exit;
+		res = fd;
```

Comment:
In case of error and in case `ENC_FS` is not defined, it will return -1, which is *not* a TEE_Result. It is then quite difficult to know the consequences after this call.